exchange/bidder.go

```diff
@@ -528,6 +532,19 @@ func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.Re
 	}
 	bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))
 	httpCallStart := time.Now()
+
+	if tmaxAdjustments != nil && tmaxAdjustments.Enabled && tmaxAdjustments.BidderResponseDurationMin > 0 {
+		if deadline, ok := ctx.Deadline(); ok {
```

Comment:
`PBSResponsePreparationDuration`  is used to calculate `enforced_tmax` duration.  This duration eventually sets the context deadline in  `makeAuctionContext()`. 

https://github.com/prebid/prebid-server/blob/d54d22b9266b2104ebbc35a586e5774ef5f61547/exchange/exchange.go#L632-L649

context prepared by `makeAuctionContext()` is then passed to `GetAllBids()` and then passed to all bidder go routines spawned from `GetAllBids()`. `PBSResponsePreparationDuration` was subtracted in `makeAuctionContext()` to avoid updating context in individual go routines.

`BidderNetworkLatencyBuffer` is not used in `enforced_tmax`. Note that `enforced_tmax` is duration PBS will wait for response and `remaining enforced_tmax - BidderNetworkLatencyBuffer` gives us `bidder_tmax` i.e duration bidder have to respond back. So these values where separated out.